File: samples/AdventureWorksLT.CSharp/AdventureWorksLT/UdtOrderNumber.cs
using DevZest.Data;
using DevZest.Data.SqlServer;
using DevZest.Data.Annotations.Primitives;
using System;
using DevZest.Data.Addons;

namespace DevZest.Samples.AdventureWorksLT
{
    [ModelDesignerSpec(addonTypes: new Type[] { typeof(ColumnNotNull), typeof(SqlType) }, validOnTypes: new Type[] { typeof(_String) })]
    public sealed class UdtOrderNumber : ColumnAttribute
    {
        protected override void Wireup(Column column)
        {
            if (column is _String stringColumn)
            {
                stringColumn.Nullable(true);
                stringColumn.AsSqlNVarChar(25);
            }
        }
    }
}

File: samples/AdventureWorksLT.Data.SqlServer.VisualBasic/AdventureWorksLT.Data/UdtOrderNumber.vb
Imports DevZest.Data.Annotations.Primitives
Imports DevZest.Data.Addons

<ModelDesignerSpec(New Type() {GetType(ColumnNotNull), GetType(SqlType)}, New Type() {GetType(_String)})>
Public NotInheritable Class UdtOrderNumber
    Inherits ColumnAttribute

    Protected Overrides Sub Wireup(ByVal column As Column)
        If TypeOf column Is _String Then
            column.Nullable(True)
            CType(column, _String).AsSqlNVarChar(25)
        End If
    End Sub
End Class
